### ISSUE TITLE
Work Checklist

### ISSUE THREAD

**Issue by Raragyay** [Raragyay/snitch_cluster#2]
GEMM:
- [ ] Verify
- [ ] Single core implementation
- [ ] Multicore implementation
- [ ] DMA
- [ ] SSR/FREP
- [ ] Tiling
- [x] Single core implementation (backprop)
- [x] Multicore implementation (backprop)
- [x] DMA (backprop)
- [x] SSR/FREP (backprop)
- [x] Tiling (backprop)

Maxpool:
- [x] Verify (all)
- Single core implementation
  - [x] 1D
  - [x] 2D
  - [x] 3D
- Multicore implementation
  - [x] 1D
  - [x] 2D
  - [x] 3D
- [x] DMA (all)
- SSR/FREP
  - [x] 1D
  - [x] 2D
  - [ ] 3D
- Tiling
  - [ ] 1D
  - [ ] 2D
  - [ ] 3D
- Return indices
  - [x] 1D
  - [x] 2D
  - [x] 3D
- Single core implementation (backprop)
  - [ ] 1D
  - [ ] 2D
  - [ ] 3D
- Multicore implementation (backprop)
  - [ ] 1D
  - [ ] 2D
  - [ ] 3D
- [ ] DMA (backprop) (all)
- SSR/FREP (backprop)
  - [ ] 1D
  - [ ] 2D
  - [ ] 3D
- Tiling (backprop)
  - [ ] 1D
  - [ ] 2D
  - [ ] 3D

Batchnorm:
Eval mode: 
- [x] Verify
- [x] Single core implementation
- [x] Multicore implementation
- [x] DMA
- [x] SSR/FREP
- [x] Tiling
- [x] Single core implementation (backprop)
- [x] Multicore implementation (backprop)
- [x] DMA (backprop)
- [x] SSR/FREP (backprop)
- [x] Tiling (backprop)

training mode: 
- [x] Verify
- [x] Single core implementation
- [x] Multicore implementation
- [x] DMA
- [x] SSR/FREP
- [ ] Tiling
- [x] Single core implementation (backprop)
- [x] Multicore implementation (backprop)
- [x] DMA (backprop)
- [x] SSR/FREP (backprop)
- [ ] Tiling (backprop)
